src/overlays/actors/ovl_Item_Ocarina/z_item_ocarina.c

```diff
@@ -27,27 +37,159 @@ const ActorInit Item_Ocarina_InitVars = {
     (ActorFunc)ItemOcarina_Update,
     (ActorFunc)ItemOcarina_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/func_80B86130.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/ItemOcarina_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/ItemOcarina_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/func_80B862EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/func_80B86494.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/func_80B864EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/func_80B865E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/func_80B86640.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/func_80B8664C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/func_80B866BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/ItemOcarina_Update.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Ocarina/ItemOcarina_Draw.s")
+void ItemOcarina_SetupAction(ItemOcarina* this, ItemOcarinaActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ItemOcarina_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ItemOcarina* this = THIS;
+    s32 params = (s32)thisx->params;
+
+    ActorShape_Init(&this->actor.shape, 0, 0, 0);
+    Actor_SetScale(&this->actor, 0.1f);
+    switch (params) {
+        case 0:
+            ItemOcarina_SetupAction(this, func_80B86494);
+            break;
+        case 1:
+            ItemOcarina_SetupAction(this, func_80B865E0);
+            break;
+        case 2:
+            ItemOcarina_SetupAction(this, func_80B86640);
+            break;
+        case 3:
+            ItemOcarina_SetupAction(this, func_80B866BC);
+            if ((gSaveContext.eventChkInf[8] & 1) == 0 || (gSaveContext.eventChkInf[4] & 8) != 0) {
+                Actor_Kill(thisx);
+                return;
+            }
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ELF_MSG2, 299.0f, -140.0f, 884.0f, 0, 4, 1, 14336);
```

Comment:
These params are more likely supposed to be hex over decimal, for now at least.
```suggestion
            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ELF_MSG2, 299.0f, -140.0f, 884.0f, 0, 4, 1, 0x3800);
```